read_mars/mstatusdisplay_to_dict.py

```diff
@@ -0,0 +1,143 @@
+from collections import OrderedDict
+import numpy as np
+import ROOT
+
+import fact
+
+
+def mstatusdisplay_to_dict(
+        path,
+        transform=True,
+        exclude=['TF1', 'TPaveText'],
+        ):
+    ''' return a flat dict, with string keys and values either coming:
+     * from an MHCamera -> np.array(shape=(1440,)) or
+     * from a TH1 -> {'bins': np.array, 'content':np.array}
+
+    keep low  level acces to the Root classes with `transform=False`
+    and more class names to the `classes` list in case you want ot
+    '''
+
+    tfile = ROOT.TFile(path)
+    status_array = tfile.Get('MStatusDisplay')
+    canvases = canvases_from_status_array(status_array)
+
+    contents = OrderedDict()
+    for name, canvas in canvases.items():
+        contents[name] = primitives_from_canvas(canvas)
+    contents = flatten_ordered_dicts(contents)
+
+    if transform:
+        output = {}
+        for name, obj in contents.items():
+            clsname = obj.__class__.__name__
+            if clsname in exclude:
+                continue
+            if clsname == 'MHCamera':
+                output[name] = transform_MHCamera(obj)
+            elif clsname in ['TH1F', 'TH1D']:
+                output[name] = transform_TH1(obj)
+            else:
+                output[name] = obj
+        return output
+    else:
+        return contents
+
+
+def transform_TH1(h):
```

Comment:
I think it makes sense to return **exactly** the same as [`np.histogram`](https://docs.scipy.org/doc/numpy/reference/generated/numpy.histogram.html) returns, i.e. a *tuple* of ("hist", "bin_edges"). 

However, I like people to be able to **guess** what the returned stuff means, without looking at the code. And one simply cannot guess what each array in a tuple of arrays could be.

If the **key** somehow conveys that the content **should be a histogram**, then one can guess what is what, because the `bin_edges` are one element longer than the `hist`. That's cool.

Or, we could return namedtuple. This is very close to the original behaviour returning just a tuple, but the names give the user a nice hint to what is going on.

----

Another thing, the bugs me is: The under- and overflow bins of ROOT histograms. I think it is a very nice feature to have these overflow bins. As they give you a quick hint if you messed out the bins, in case you defined them manually. In this case where we have a ROOT histogram to begin with and convert it into a python structure, we would thrown existing information away, if we simply ignored the overflow bins, which I do not like very much.

So I would propose so to return `bin_edges` like this:

```
>>> np.random.seed(1)
>>> r = np.random.normal(1, 1, 10)
>>> np.histogram(r, bins=[0, 1, 2, 3])
# Out: (array([4, 2, 2]), array([0, 1, 2, 3]))   <-- 2 numbers are out of range

# This is how root does it
>>> np.histogram(r, bins=[-np.inf, 0, 1, 2, 3, np.inf])  # simply add two bins with infinite size
#Out: (array([2, 4, 2, 2, 0]), array([-inf,   0.,   1.,   2.,   3.,  inf]))  # the sum of `hist` is 10 ... awesome
```

So I would propose to simply append two bin edges and also return the under- and overflow bins as if they were just normal bins. 

----

I just spoke with Sebastian about this ... and he argues, that this can be even more surprising. We agreed, that the interface of `np.histogram` is anyway broken and not worth to be conserved. So maybe a named tuple like:

```
Hist = namedtuple('Hist', "bin_edges, bin_contents, underflowbin, overflowbin") 
```

would be a better return value. Now that I write this, I remember we once had code somewhere, that provided some useful helper functions to quickly calc some statistics on Histograms ... do we still have that? maybe we can use it here?